installer/kickstart.sh

```diff
@@ -297,7 +302,7 @@ then
         if [ "${ask}" = "1" ]
         then
             echo >&2 "It failed to install all the required packages, but I can try to install netdata."
-            read -p "Press ENTER to continue to netdata installation > "
+            read -r -p "Press ENTER to continue to netdata installation > "
```

Comment:
`-r` works on busybox too, so this is ok.

Reply:
Where is busybox used? I mean "who" uses it and for what? I am just curious.

Reply:
alpine linux

Reply:
all IoT

Reply:
On alpine:

```sh
# ls -l /bin/
total 2060
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ash -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 base64 -> /bin/busybox
-rwxr-xr-x    1 root     root        714912 Feb  7  2018 bash
-r-xr-xr-x    1 root     root          6971 Feb  7  2018 bashbug
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 bbconfig -> /bin/busybox
---s--x--x    1 root     root         10016 Jun  6 09:09 bbsuid
-rwxr-xr-x    1 root     root        792808 Jun  6 09:09 busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 cat -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 chgrp -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 chmod -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 chown -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 conspy -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 cp -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 date -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 dd -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 df -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 dmesg -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 dnsdomainname -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 dumpkmap -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 echo -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ed -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 egrep -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 false -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 fatattr -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 fdflush -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 fgrep -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 fsync -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 getopt -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 grep -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 gunzip -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 gzip -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 hostname -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ionice -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 iostat -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ipcalc -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 kbd_mode -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 kill -> /bin/busybox
-rwxr-xr-x    1 root     root        124880 Oct 31  2017 kmod
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 link -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 linux32 -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 linux64 -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ln -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 login -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ls -> /bin/busybox
lrwxrwxrwx    1 root     root             4 Jun 22 00:29 lsmod -> kmod
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 lzop -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 makemime -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 mkdir -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 mknod -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 mktemp -> /bin/busybox
lrwxrwxrwx    1 root     root             4 Jun 22 00:29 modinfo -> kmod
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 more -> /bin/busybox
lrwxrwxrwx    1 root     root            11 Sep 11 14:40 mount -> /bin/bbsuid
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 mountpoint -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 mpstat -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 mv -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 netstat -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 nice -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 pidof -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ping -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ping6 -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 pipe_progress -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 printenv -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 ps -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 pwd -> /bin/busybox
-rwxr-xr-x    1 root     root         26504 Nov 28  2017 rc-status
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 reformime -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 rev -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 rm -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 rmdir -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 run-parts -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 sed -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 setpriv -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 setserial -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 sh -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 sleep -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 stat -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 stty -> /bin/busybox
lrwxrwxrwx    1 root     root            11 Sep 11 14:40 su -> /bin/bbsuid
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 sync -> /bin/busybox
-rwxr-xr-x    1 root     root        402472 Oct 31  2017 tar
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 touch -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 true -> /bin/busybox
lrwxrwxrwx    1 root     root            11 Sep 11 14:40 umount -> /bin/bbsuid
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 uname -> /bin/busybox
-rwxr-xr-x    1 root     root         14040 Mar  5  2018 uniso
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 usleep -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 watch -> /bin/busybox
lrwxrwxrwx    1 root     root            12 Jun 21 19:00 zcat -> /bin/busybox
```

haha...

Reply:
I need to do some research on Azure CI/CD, maybe they allow more OSes than just ubuntu 14.04 on travis. This way we could use it in parallel with travis and build on actual supported systems.

Reply:
nice!

Keep in mind I run containers of almost every version of Linux at http://stackscale.my-netdata.io/
This is a server donated by stackscale and it runs the cron job to build the static build, daily.

If you need it, I'll be glad to give you access.

Also stackscale, is pretty keen to offer us all the VMs and bandwidth we need to host binary packages for netdata.

Reply:
Can you link to scripts you use for that? I could probably integrate them into travis pipelines.

Reply:
Sure, here they are: https://github.com/firehol/binary-packages

Reply:
oh... you mean to create the containers?

They are here: https://github.com/firehol/netdata-demo-site/tree/master/lxc

Reply:
oh...lxc... nevermind I'll do that in docker :)

Reply:
haha....

Reply:
Here we go: https://github.com/firehol/netdata/pull/4174 :)